pkg/auth/tokenauthenticator.go

```diff
@@ -38,3 +59,94 @@ func (t *tokenAuthenticator) AddAuthenticationHeader(request *http.Request) erro
 func (t *tokenAuthenticator) IsSupported() bool {
 	return true
 }
+
+func IsAuthTypeToken(token string) bool {
+	if _, uuidErr := uuid.Parse(token); uuidErr == nil {
+		return true
+	}
+	return false
+}
+
+func IsAuthTypePAT(token string) bool {
+	// e.g. snyk_uat.1a2b3c4d.mySuperSecret_Token-Value.aChecksum_123-Value
+	patRegex := `^snyk_(?:uat|sat)\.[a-z0-9]{8}\.[a-zA-Z0-9-_]+\.[a-zA-Z0-9-_]+$`
+	if matched, err := regexp.MatchString(patRegex, token); err == nil && matched {
+		return matched
+	}
+	return false
+}
+
+// DeriveEndpointFromPAT makes an authenticated request to the Snyk PAT API and returns the correct Snyk endpoint from the given PAT
+var DeriveEndpointFromPAT = func(token string, config configuration.Configuration, client *http.Client, patServiceUrl string) (string, error) {
+	// TODO: we're implementing a local caching solution here but this will need a proper caching solution
+	// exit early if pat is already cached
+	cached_pat_key := GetCachedPatKeyname(token)
+	cached_pat := config.GetString(cached_pat_key)
+	if len(cached_pat) > 0 {
+		return cached_pat, nil
```

Comment:
might be just naming, but where exactly is the endpoint returned in this case?

Reply:
Yeah, I think it is a naming issue 😆